### PR TITLE
Add DR.Kernel recipe: Triton kernel-generation RL on NPU

### PR DESCRIPTION
**Pull request by HwCARI**
# [recipe] feat: add Dr. Kernel asynchronous RL recipe for Triton kernel generation on Ascend NPU

## What does this PR do?

This PR adds a new `drkernel` recipe that brings the **Dr. Kernel** approach (RL Done Right for Triton kernel generation) to `verl-ascend-recipe`, adapted to run on Ascend NPU using **fully asynchronous RL training**. The model is given a Torch reference implementation and learns to produce optimized kernel code, with execution-based verification.

## Background

Dr. Kernel targets two failure modes common to RL for kernel generation — reward hacking and lazy optimization — using the KernelGym execution environment for correctness/performance checks, profiling-based rewards, and unbiased multi-turn RL. This recipe ports that workflow onto the verl + Ascend stack.

To address this, the recipe is built on verl's **fully asynchronous training** path:

- The **trainer** and **rollout/evaluation** workers are fully decoupled and run concurrently.
- Rollouts and kernel verification stream results back through an async queue, so the trainer consumes samples as they complete instead of waiting on the slowest kernel.
- Training proceeds **off-policy** with controlled staleness, keeping device utilization high during long-horizon, multi-turn rollouts.

This asynchronous design is the key enabler for scaling Dr. Kernel-style RL efficiently on Ascend NPU.

## Changes

- New `drkernel/` recipe directory containing:
  - `README.md` — setup, data prep, and async training instructions for Ascend NPU
  - `REQUIRED_VERL.txt` — pinned/rolling verl version per repo convention
  - Fully asynchronous training scripts and config for [model, e.g. Qwen3-8B] multi-turn GRPO RL (decoupled trainer / rollout, off-policy with staleness control)
  - Reward functions covering compilation, correctness, and performance
  - Integration with the kernel evaluation environment through the async rollout/reward pipeline
- Ascend-specific adaptations (triton-ascend, NPU device handling, async-mode env setup)

### CODE REVIEW

**Comment by HwCARI** [comment]
Do you have any comments regarding this PR to be addressed to facilitate merging this recipe?

**Review by wucong25 — APPROVED** [review]

File: drkernel/agent/kernel_agent_loop.py [hunk 1]
```diff
@@ -0,0 +1,600 @@
+"""DR.Kernel-style multi-turn agent loop on top of verl's `AgentLoopBase`.
```
Comment (wucong25) [inline drkernel/agent/kernel_agent_loop.py:3]: Need to add Huawei copyright in all the files
